File: courses/codes/hw3-py-sol/hw3_q7.py
n = int(input("Please enter integers:\n"))

count_odd = 0           # counts how many odd integers are inputted
sum_odd = 0             # sums the values of all odd integers

while n != 0:
    if n % 2 == 1:      # checks if integer is odd
        count_odd += 1  # counts the odd integers
        sum_odd += n    # adds the odd integer to the cumulative sum
    n = int(input())    # asks the integer at each iteration; only stop when n = 0

if count_odd == 0:
    print(0)
else:
    print(sum_odd / count_odd)

File: courses/codes/java-class/hw3-java-sol/HW3Q7.java
import java.util.Scanner;

public class HW3Q7 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int countOdd = 0;		// counts all odd integers seen so far
		int sumOdd = 0;			// adds all odd integers seen so far
		
		System.out.println("Please enter integers: ");
		int n = sc.nextInt();
		
		while (n != 0) {
			
			int absN = (n >= 0) ? n : -n;
			
			if (absN % 2 == 1) {		// checks if absolute n is odd
				countOdd++;
				sumOdd += n;
			}
			
			n = sc.nextInt();
		}
		
		if (countOdd == 0) {
			System.out.println(0);
		} else {
			double average = (double) sumOdd / countOdd;
			System.out.println(average);
		}
	}

}
